src/lib.rs

```diff
@@ -695,7 +695,8 @@ where
     {
         let addr: AddrKind = addr.into();
         let bytes = Self::write_request(buf, addr).await?;
```

Comment:
I think it would be better to return header length from `write_request`

Reply:
I've updated that.